src/git.js

```diff
@@ -208,6 +208,33 @@ module.exports = function gitFactory(repository = '', origin = 'origin') {
                 .then( () => git(repository).push(origin, baseBranch) );
         },
 
+        /**
+         * Push a branch to the given remote
+         * @param {String} origin - name of the remote
+         * @param {String} branchName - name of the branch to push to
+         * @returns {Promise}
+         */
+        push(origin, branchName) {
```

Comment:
async missing

Reply:
on the contrary, because the `await` will be for the returned promise... outside of the function

I will remove `async` from `merge` and `abortMerge` methods.